### PR TITLE
[FLINK-25916][connector-kafka] Using upsert-kafka with a flush buffer…

### PR DESCRIPTION
**Pull request by paul8263**
… results in Null Pointer Exception

This PR is the migration from [FLINK-25916](https://github.com/apache/flink/pull/20343)

# What is the purpose of the change

Allow flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriter.WrappedContext to set timestamp with NULL value in order to avoid NPE.

# Brief change log

- flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriter.java
- flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest.java

# Verifying this change

This change is already covered by existing tests, such as flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest::testWriteDataWithNullTimestamp().

# Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency): no
- The public API, i.e., is any changed class annotated with @Public(Evolving): no
- The serializers: no
- The runtime per-record code paths (performance sensitive): no
- Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
- The S3 file system connector: no

# Documentation

- Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by mhv666** [comment]
Hi @jaumebecks, I'm waiting as well for any updates 🙏🏽 Thanks guys

**Comment by tzulitai** [comment]
Hey folks @paul8263 @jaumebecks @mhv666, I can try to review this next week after some other work - will need a bit of time to catch up. Thanks for the patience.

**Review by Gerrrr — COMMENTED** [review]
The change makes sense to me. 

In addition to the unit test, it would be beneficial to add a test case to `UpsertKafkaTableITCase` that shows the use case where the timestamp is null.

**Comment by tzulitai** [comment]
Changes LGTM, +1 to @Gerrrr's comment on extending `UpsertKafkaTableITCase` before we merge this.

**Comment by paul8263** [comment]
Hi @tzulitai ，
Thank you for reviewing the code.

After adding some tests in `UpsertKafkaTableITCase` I got some exceptions like: javax.management.InstanceAlreadyExistsException.

As GitHub CI page scroll is laggy, I pushed the same codes to Flink project and got hive-connector compilation failure, which is weird.

How could I rerun the unit tests and solve those errors? Thanks.

**Comment by tzulitai** [comment]
@paul8263 I think you can just go to the CI job page and re-run the job.

**Comment by paul8263** [comment]
Hi @tzulitai ，

Could you please help approve the workflow? It seems that the CI process won't run anymore.

I tried pushing the same commit to both this project and the legacy Flink project, but I got different errors. May take some time to figure it out.

**Comment by MartijnVisser** [comment]
> Could you please help approve the workflow? It seems that the CI process won't run anymore.

The new ASF guidelines require approval from a committer to run a CI flow; I've just started it for you

**Review by tzulitai — APPROVED** [review]
Thanks for opening the PR and addressing review comments @paul8263!

LGTM, merging this now.

**Pull request closed without merging**
